chainntnfs/txnotifier.go

```diff
@@ -1243,6 +1270,36 @@ func (n *TxNotifier) dispatchConfReorg(ntfn *ConfNtfn,
 	return nil
 }
 
+// dispatchSpendReorg dispatches a reorg notification to the client if a spend
+// notiification was already delivered.
+//
+// NOTE: This must be called with the TxNotifier's lock held.
+func (n *TxNotifier) dispatchSpendReorg(ntfn *SpendNtfn) error {
+	if !ntfn.dispatched {
+		return nil
+	}
+
+	// Attempt to drain the spend notification to ensure sends to the Spend
+	// channel are always non-blocking.
+	select {
+	case <-ntfn.Event.Spend:
+	default:
+	}
+
+	// Send a reorg notification to the client in order for them to
+	// correctly handle reorgs.
+	select {
+	case ntfn.Event.Reorg <- struct{}{}:
+	case <-n.quit:
+		return ErrTxNotifierExiting
+	}
+
+	ntfn.dispatched = false
+	n.spendNotifications[ntfn.OutPoint].details = nil
```

Comment:
this is already done before calling `dispatchSpendReorg`, no?

Reply:
Yep, removed from here.